site/src/pages/ChatPage/ChatLayout.tsx

```diff
@@ -0,0 +1,246 @@
+import { useTheme } from "@emotion/react";
+import AddIcon from "@mui/icons-material/Add";
+import Button from "@mui/material/Button";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemText from "@mui/material/ListItemText";
+import Paper from "@mui/material/Paper";
+import { createChat, getChats } from "api/queries/chats";
+import { deploymentLanguageModels } from "api/queries/deployment";
+import type { LanguageModelConfig } from "api/typesGenerated";
+import { ErrorAlert } from "components/Alert/ErrorAlert";
+import { Loader } from "components/Loader/Loader";
+import { Margins } from "components/Margins/Margins";
+import { useAgenticChat } from "contexts/useAgenticChat";
+import {
+	type FC,
+	type PropsWithChildren,
+	createContext,
+	useContext,
+	useEffect,
+	useState,
+} from "react";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { Link, Outlet, useNavigate, useParams } from "react-router-dom";
+
+export interface ChatContext {
+	selectedModel: string;
+	modelConfig: LanguageModelConfig;
+
+	setSelectedModel: (model: string) => void;
+}
```

Comment:
review: Ideally I'd like to move this into `useAgenticChat` but ran out of time.